nfs/src/rpc_task.cpp

```diff
@@ -1072,13 +1089,25 @@ void rpc_task::run_write()
      * membufs. We don't wait for the writes to actually finish, which means
      * we support buffered writes.
      */
-    const std::vector<bytes_chunk> bc_vec =
-        copy_to_cache(get_client(), ino, bufv, offset, length, error_code);
+    std::vector<bytes_chunk> bc_vec =
+            copy_to_cache(this, ino, bufv, offset, length, error_code);
+    if (error_code != 0) {
+        reply_error(error_code);
+        return;
+    }
+
     assert(bc_vec.size() > 0);
 
-    for (const bytes_chunk& bc : bc_vec) {
-        // Flush the membuf to backend.
-        sync_membuf(bc, get_client(), ino);
+    for (bytes_chunk& bc : bc_vec) {
+        /*
+         * Create the flush task to carry out the write.
+         */
+        struct rpc_task *flush_task =
+            get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
+        flush_task->init_flush(nullptr /* fuse_req */, rpc_api->flush_task.get_ino());
+        flush_task->rpc_api->pvt = nullptr;
```

Comment:
instead of doing it here we should do it in api_task_info::release().
Here we can assert